server/src/main/java/org/elasticsearch/search/aggregations/metrics/MaxAggregator.java

```diff
@@ -127,6 +128,33 @@ public InternalAggregation buildEmptyAggregation() {
         return Max.createEmptyMax(name, formatter, metadata());
     }
 
+    @Override
+    public boolean canUseCollectedAggregator() {
+        return true;
+    }
+
+    @Override
+    public CollectedAggregator buildCollectedAggregator(long[] ordsToCollect) {
+        CollectedMax collectedMax = new CollectedMax(name, metadata(), bigArraysForResults(), ordsToCollect.length, formatter);
```

Comment:
It's almost definitely wrong to be creating a new `CollectedAggregator` here. We want there to be exactly one of these per aggregator.